### DISCUSSION TITLE
Gradient Descent Visualization with PyVista

### DISCUSSION THREAD

**Discussion by JacobBumgarner** [pyvista/pyvista#3074] [Show and tell]
Hi all, I wanted to show off a small package that I made when learning about the fundamentals of gradient descent.

https://user-images.githubusercontent.com/70919881/180077858-14bd8b91-c189-4e52-80d4-332ae5ca2db4.mov

The package is called [grad-descent-visualizer](https://github.com/JacobBumgarner/grad-descent-visualizer). It comes with a variety of example test functions and several visualization and movie generation features. It's mostly an educational package that I built to further my understanding of gradient descent from first principles.

https://user-images.githubusercontent.com/70919881/180919957-15773f3d-0ac1-4ce2-a1e5-8cfd9a624c24.mov

Importantly, I built the package with PyVista. The package plotter is built on top of the PyVista Plotter, which uses Uniform Grids to visualize test function landscapes, Glyphs to visualize function gradient orientations and points as they descend their gradients, and Tube-filtered PolyData to visualize the paths of points during their descents. Examples of package use can be found on the [repo's README](https://github.com/JacobBumgarner/grad-descent-visualizer#grad-descent-visualizer).

https://user-images.githubusercontent.com/70919881/180921507-d47b2e75-b512-4691-9496-c75510e4158b.mov

I also featured this package and PyVista in a recent [Medium post](https://medium.com/@jacobbumgarner/breaking-it-down-gradient-descent-b94c124f1dfd).

https://user-images.githubusercontent.com/70919881/180920048-06c76ba3-2cf2-43aa-ad74-6629953ceb64.mov

**Comment by Onandon11** [comment]
Really cool! Very nice visualization! 

Maybe this could even be included if we implemented #3184?

**Reply by banesullivan** [reply]
Awesome stuff!! I love seeing these "show and tell" style posts!
